migrations/dimensions-config/bf6308af80aa.py

```diff
@@ -0,0 +1,402 @@
+    likely we never need it.
+    """
+    raise NotImplementedError()
+
+
+def _FKConstraint(
+    src_table: str,
+    src_columns: List[str],
+    tgt_table: str,
+    tgt_columns: List[str],
+    bind: sa.engine.Connection,
+    schema: Optional[str] = None,
+    ondelete: Optional[str] = None,
+) -> sa.schema.ForeignKeyConstraint:
+    """Create foreign key constraint."""
+    fk_name = "_".join(["fkey", src_table, tgt_table] + tgt_columns + src_columns)
+    fk_name = shrinkDatabaseEntityName(fk_name, bind)
+    tgt_columns = [f"{tgt_table}.{col}" for col in tgt_columns]
+    if schema:
+        tgt_columns = [f"{schema}.{col}" for col in tgt_columns]
+    return sa.schema.ForeignKeyConstraint(src_columns, tgt_columns, name=fk_name, ondelete=ondelete)
+
+
+def _migrate_visit_definition() -> None:
+    """Make new visit_system_membership table, fill from existing data
+    in visit_definition."""
+
+    mig_context = context.get_context()
+    bind = mig_context.bind
+    schema = mig_context.version_table_schema
+    metadata = sa.schema.MetaData(bind, schema=schema)
+
+    table = "visit_system_membership"
+    _LOG.info("creating %s table", table)
+    constraints = [
+        _FKConstraint(table, ["instrument"], "instrument", ["name"], bind, schema=schema),
+        _FKConstraint(
+            table, ["instrument", "visit_system"], "visit_system", ["instrument", "id"], bind, schema=schema
+        ),
+        _FKConstraint(table, ["instrument", "visit"], "visit", ["instrument", "id"], bind, schema=schema),
+    ]
+    if bind.dialect.name == "sqlite":
+        constraints.append(
+            sa.schema.CheckConstraint(
+                "length(instrument)<=16 AND length(instrument)>=1",
+                name=shrinkDatabaseEntityName("_".join([table, "len", "instrument"]), bind),
+            )
+        )
+    visit_membership = op.create_table(
+        table,
+        sa.Column("instrument", sa.String(16), primary_key=True, autoincrement=False),
+        sa.Column("visit_system", sa.BigInteger, primary_key=True, autoincrement=False),
+        sa.Column("visit", sa.BigInteger, primary_key=True, autoincrement=False),
+        *constraints,
+        schema=schema,
+    )
+
+    op.create_index(
+        shrinkDatabaseEntityName("_".join([table, "fkidx", "instrument", "visit_system"]), bind),
+        table,
+        ["instrument", "visit_system"],
+        schema=schema,
+    )
+    op.create_index(
+        shrinkDatabaseEntityName("_".join([table, "fkidx", "instrument", "visit"]), bind),
+        table,
+        ["instrument", "visit"],
+        schema=schema,
+    )
+    op.create_index(
+        shrinkDatabaseEntityName("_".join([table, "fkidx", "instrument"]), bind),
+        table,
+        ["instrument"],
+        schema=schema,
+    )
+
+    visit_definition = sa.schema.Table("visit_definition", metadata, autoload_with=bind, schema=schema)
+
+    # Copy ever0thing from visit_definition to the new table
```

Comment:
Typo: "ever0thing"

Also, python PEP8 really wants a period at the end of comments and capital letters at the start... (full sentences seemingly)